Dataset <--> pandas DataFrame

JasperVanDenBosch opened the pull request:
Proposal for DataFrame conversion of Datasets. This would make it a bit easier to write code for importing table-like data files, such as those stored in event files, CSVs.


### example
```python
ds_in = Dataset(
    measurements=numpy.random.rand(3, 2),
    descriptors=dict(foo='bar'),
    obs_descriptors=dict(abc=['a', 'b', 'c']),
    channel_descriptors=dict(name=['x', 'y'])
)
df = ds_in.to_DataFrame()
ds_out = Dataset.from_DataFrame(df)
print(df)
```
```
          x         y         abc  foo
0  0.709961  0.514268           a  bar
1  0.494080  0.100730           b  bar
2  0.922560  0.530450           c  bar
```

### Related 
- #160 
- #80

JasperVanDenBosch committed:
new dependency: pandas

JasperVanDenBosch committed:
dataset <-> dataframe: method signatures and acceptance test

JasperVanDenBosch committed:
implemented Dataset.to_DataFrame()

JasperVanDenBosch committed:
implemented Dataset.from_DataFrame()

JasperVanDenBosch committed:
[style] removed trailing whitespace

JasperVanDenBosch committed:
simple paragraph on pandas dataframes

HeikoSchuett reviewed (commented):
In general this looks good to me. 

Two comments: 

1. Can you add an input to to_DataFrame that allows one to choose which descriptor to use for the channel names? I think that would be a good addition
2. Similarly, I think it would be great if the from_DataFrame function accepted an argument to decide which columns should be come descriptors and which ones should not. People being brought up with numeric matrices will often produce descriptors that are numbers and it might also become frustrating to have integer data not accepted or something like this. Floats are data is a good heuristic default, but not perfect.

JasperVanDenBosch committed:
dataframe conversion methods take optional channel args

JasperVanDenBosch committed:
[style] changed method names

JasperVanDenBosch commented:
@HeikoSchuett Thanks for looking at this! I've added the arguments. I've also changed the method names (`to_df()`, `from_df()`) since codefactor didn't like them (and I suppose this way they are more in line with e.g. `to_dict()`), is this good?

HeikoSchuett commented:
I understand that it is generally against doing both camel case and underscores in a name. Also, it is now shorter. I'll just merge this in then!
